tests/test_giant.py

```diff
@@ -0,0 +1,171 @@
+import unittest
+import numpy as np
+import numpy.testing as npt
+from squirrel import giant
+from squirrel.data import Spectra
+from squirrel.data import Datacube
+from squirrel.template import Template
+
+
+class TestShoulder(unittest.TestCase):
+    def setUp(self):
+        """Set up the test."""
+        self.wavelengths = np.arange(1, 100)
+        self.flux = np.ones_like(self.wavelengths)
+        self.flux_unit = "arbitrary unit"
+        self.wavelength_unit = "nm"
+        self.noise = np.ones_like(self.flux)
+        self.fwhm = 2.0
+        self.z_lens = 0.5
+        self.z_source = 1.0
+        self.spectra = Spectra(
+            self.wavelengths,
+            self.flux,
+            self.wavelength_unit,
+            self.fwhm,
+            self.z_lens,
+            self.z_source,
+            self.flux_unit,
+            self.noise,
+        )
+
+    def test_log_rebin(self):
+        giant.Shoulder.log_rebin(self.spectra)
+        self.assertIn("log_rebinned", self.spectra.spectra_modifications)
+
+    def test_voronoi_bin(self):
```

Comment:
For tests that require specific setup choices (such as the number of pixels), I personally like using `pytest` with the decorators `@pytest.mark.parametrize`, that allows to parametrize the test function to reduce code duplication. For instance, here this could be parametrize on the number of grid pixels, for example to test even vs odd number of pixels on each side.
Just a thought, there are perhaps advantages of `unittest` over `pytest` that I am not aware of.

Reply:
Thanks for the suggestion. I have changed the testing framework to `pytest`. 🙂